manifests/overlays/dev/mock-server.yaml

```diff
@@ -0,0 +1,138 @@
+# This YAML adds a mock server of gpu metrics for gpu testing
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nvidia-dcgm-exporter
+  labels:
+    app: nvidia-dcgm-exporter
+spec:
+  replicas: 1
+  selector:
+    type: RollingUpdate
+    matchLabels:
+      app: nvidia-dcgm-exporter
+  template:
+    metadata:
+      labels:
+        app: nvidia-dcgm-exporter
+    spec:
+      containers:
+        - name: metrics
+          image: registry.redhat.io/rhel8/nodejs-14-minimal:latest
+          imagePullPolicy: Always
+          workingDir: /usr/tmp
+          command:
+            - /bin/bash
+            - -c
+            - |
+              #/bin/bash
+              cat <<EOF>> package.json
+              {
+                  "type": "module"
+              }
+              EOF
+              cat <<EOF > mock-server.js
+              import client from 'prom-client';
+              import http from 'http';
+              import url from 'url';
+
+              const registry = new client.Registry
+
+              const mockGPU = new client.Gauge(
+                  {
+                      name: "DCGM_FI_PROF_GR_ENGINE_ACTIVE",
+                      help: "Mock gpu data",
+                      labelNames: [
+                          'Hostname',
+                          'UUID',
+                          'device',
+                          'gpu',
+                          'instance',
+                          'endpoint',
+                          'namespace',
+                          'modelName',
+                      ],
+                      registers: [registry],
+                  },
+              );
+
+              ` DCGM_FI_PROF_GR_ENGINE_ACTIVE
+              {
+                  Hostname="ip-10-0-150-8",
+                  UUID="GPU-cd248e1c-1c37-7bd5-076c-25d95eca8e2b", 
+                  container="nvidia-dcgm-exporter", 
+                  device="nvidia0", 
+                  endpoint="gpu-metrics", 
+                  gpu="0", 
+                  instance="10.0.150.8:9400", 
+                  job="nvidia-dcgm-exporter", 
+                  modelName="Tesla T4", 
+                  namespace="nvidia-gpu-operator", 
+                  pod="nvidia-dcgm-exporter-b4f8j", 
+                  service="nvidia-dcgm-exporter"
+              }
+
+              `
+
+              mockGPU.labels(
+                  {
+                      modelName: "Tesla T4",
+                      UUID: "GPU-cd248e1c-1c37-7bd5-076c-25d95eca8e2b",
+                      device: "nvidia0",
+                      gpu: "1",
+                      instance: "10.0.150.8:9400",
+                      endpoint: "gpu-metrics",
+                      namespace: "nvidia-gpu-operator",
+                      Hostname: "ip-10-0-150-8"
+
+                  }).set(1)
+
+              mockGPU.labels(
```

Comment:
After talking with @VaishnaviHire and having a better understanding of the nvidia exporter we should be testing the following scenario:

- A node (node1.opendatahub.io) has 3 GPUs but one of them (gpu-2) is being used by a pod so the UI should only display 2 available:

```node
mockGPU.labels(
    {
        Hostname: "node1.opendatahub.io",
        UUID: "GPU-cd248e1c-1c37-7bd5-076c-25d95eca8e2b",
        device: "nvidia0",
        modelName: "Tesla T4",
        gpu: "0",
    }).set(0)

mockGPU.labels(
    {
        Hostname: "node1.opendatahub.io",
        UUID: "GPU-691bd394-0924-4ec9-9baa-c79d97f26a30",
        device: "nvidia0",
        modelName: "Tesla T4",
        gpu: "1",
    }).set(0)

mockGPU.labels(
    {
        Hostname: "node1.opendatahub.io",
        UUID: "GPU-d0c6cbdf-719e-44d6-9810-7bb62e565997",
        device: "nvidia0",
        modelName: "Tesla T4",
        gpu: "2",
        pod: "this-gpu-is-busy",
    }).set(0)
```